resources/todomvc/architecture-examples/preact/src/todo/components/item.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useCallback } from "preact/hooks";
+import { memo } from "preact/compat";
+
+import { Input } from "./input";
+
+export const Item = memo(function Item({ todo, dispatch }) {
+    const [isWritable, setIsWritable] = useState(false);
+    const { title, completed, id } = todo;
+
+    const toggleItem = useCallback(() => dispatch({ type: "TOGGLE_ITEM", payload: { id } }), [dispatch]);
+    const removeItem = useCallback(() => dispatch({ type: "REMOVE_ITEM", payload: { id } }), [dispatch]);
+    const updateItem = useCallback((id, title) => dispatch({ type: "UPDATE_ITEM", payload: { id, title } }), [dispatch]);
+
```

Comment:
This comment applies to other similar places where `useCallback` is used too.

It's probably not necessary to use `useCallback` (and even could have negative implications to the performance of the app because of the unneeded memoization). Indeed `useCallback` is useful when you want to keep the same instance of the function. In this case it's not needed because they're used either on plain dom nodes or Components that aren't using `memo`.

Although we may want to have some of them memoized just for the sake of being closer to real-life applications? But in that case please add comments stating it explicitely.

Reply:
the Item is wrapped in a memo function

Reply:
@julienw Preact might be different in this regard (due to the lack of a synthetic event system), but passing functions to DOM nodes isn't free in React. Negligible cost? Probably. But not free. Internally, plain DOM nodes in React have a sort of memo wrapper, and changing event handlers deopts them and causes some reconciliation work that is otherwise unnecessary.